frontend/src/components/Scratch/Comments.tsx

```diff
@@ -0,0 +1,255 @@
+"use client"
+
+import { ChangeEvent, useCallback, useRef, useState } from "react"
+
+import { useRouter } from "next/navigation"
+
+import { ArrowUpIcon, KebabHorizontalIcon, CheckIcon } from "@primer/octicons-react"
+import { formatDistanceToNowStrict } from "date-fns"
+import useSWR, { mutate } from "swr"
+
+import Loading from "@/components/loading.svg"
+import * as api from "@/lib/api"
+import { Comment } from "@/lib/api"
+import { commentUrl } from "@/lib/api/urls"
+
+import AsyncButton from "../AsyncButton"
+import Button from "../Button"
+import Dropdown from "../Dropdown"
+import UserLink from "../user/UserLink"
+
+import styles from "./Comments.module.scss"
+
+const maxTextLength = 5000
+
+async function deleteComment(comment: Comment) {
+    try {
+        await api.delete_(commentUrl(comment), {})
+        alert("Comment deleted")
+    } catch (e) {
+        alert(`Error Deleting Comment: ${e}`)
+    }
+}
+
+function EditComment({ comment, stopEditing, submit }: { comment: any, stopEditing: () => void, submit: (text: string) => Promise<unknown> }) {
+    const [text, setText] = useState(comment.text)
+    const textareaRef = useRef<HTMLTextAreaElement>(null)
+
+    const handleInput = (e: ChangeEvent<HTMLTextAreaElement>) => {
+        if (textareaRef.current) {
+            textareaRef.current.style.height = "auto"
+            textareaRef.current.style.height = `${e.target.scrollHeight + 2}px`
+        }
+        setText(e.target.value)
+    }
+
+    return (
+        <div>
```

Comment:
Ideally this would be more semantic, like a form, and editing would submit the form